sklearn/gaussian_process/tests/test_gpr.py

```diff
@@ -569,3 +551,28 @@ def test_constant_target(kernel):
     assert_allclose(y_pred, y_constant)
     # set atol because we compare to zero
     assert_allclose(np.diag(y_cov), 0., atol=1e-9)
+
+
+def test_std_no_inv():
+    # Test for issue #19936
+    # Form a Kernel that is difficult to compute K_inv,
+    # necessitates cho_solve in std computation.
+    kernel = C(8.98576054e+05, (1e-12, 1e12)) * \
+        RBF([5.91326520e+02, 1.32584051e+03], (1e-12, 1e12)) + \
+        WhiteKernel(noise_level=1e-5)
```

Comment:
I realized I waste some time in the test fitting an already fit set of hyperparameters, resolved by passing `optimizer=None`

I believe the difficulties result not from the high precision, but from the fact that the generated kernel matrix is almost-not invertible.  For this specific test case I get eigenvalues
```
[5.39143784e+06 1.57542098e+01 2.73160492e+00 3.35498069e-05 1.03725624e-05 1.35603633e-05]
```
So probably a pretty poor condition number.  This matrix is nearly equal to a (scaled) matrix of all 1's, which is singular.

The kernel values I used in the minimal example are pulled directly from an issue I was having, with the values the result of `gpr.fit` When a GPR is attempting to fit data that is linearly related (as this is) the length scales  of the RBF shoot up to relatively large numbers, since the correlation length for a perfect linear fit is inf. 

If my understanding is correct: This new implementation can still experience instability, however you would be hard pressed to encounter a test dataset that exposes this instability to the same extent the previous implementation did. `solve_triangular(self.L_.T, np.eye(self.L_.shape[0]))` In the previous implementation was likely producing inverses with large differences in magnitudes between the entries. `K_trans` looking like a column of `np.eye` would be difficult with such a kernel.